OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -184,6 +184,16 @@ protected override void Created(Actor self)
 			base.Created(self);
 		}
 
+		public void AddNotifyAttacks(Actor attacker, INotifyAttack[] notifyAttacks)
+		{
+			this.notifyAttacks = this.notifyAttacks.Append(notifyAttacks.Select(a => (attacker, a)).ToArray()).ToArray();
```

Comment:
That shouldn't be needed.

```suggestion
			this.notifyAttacks = this.notifyAttacks.Append(notifyAttacks.Select(a => (attacker, a))).ToArray();
```

Reply:
it was needed

Reply:
Hm, why? `notifyAttacks` is the local variable here and not modified, while `this.notifyAttacks` is modified? 🤔

Reply:
Just append requires an array type. But we aren't using append anymore anyhow